core/src/main/java/org/jruby/ext/set/RubySet.java

```diff
@@ -875,6 +876,18 @@ public IRubyObject op_equal(ThreadContext context, IRubyObject other) {
         return context.runtime.getFalse();
     }
 
+    @JRubyMethod(name = "reset")
+    public IRubyObject reset(ThreadContext context) {
+        IRubyObject tmp = TypeConverter.checkHashType(context.runtime, this.hash);
```

Comment:
`this.hash` is 'internal' -> can not be changed (wout hacks) on .rb end ... + its already of type `RubyHash`

Reply:
Not sure if I understand correct :/ What should I use instead ``this.hash``.
The type check is adapted from the MRI implementation.

Reply:
> The type check is adapted from the MRI implementation.

figured, since this as a native part, specific to JRuby, just logically simplify what makes sense.
since you have a `RubyHash hash` you do not need any type-checking whether its a `Hash` ...